flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -625,3 +625,19 @@ bool Fortran::lower::CharacterExprHelper::isArray(mlir::Type type) {
   }
   return false;
 }
+
+bool Fortran::lower::CharacterExprHelper::hasConstantLengthInType(
+    const fir::ExtendedValue &exv) {
+  auto type = fir::getBase(exv).getType();
+  if (auto boxTy = type.dyn_cast<fir::BoxType>())
+    type = boxTy.getEleTy();
+  if (auto eleTy = fir::dyn_cast_ptrEleTy(type))
+    type = eleTy;
+  if (auto seqTy = type.dyn_cast<fir::SequenceType>()) {
+    auto lenVal = seqTy.getShape()[0];
+    assert(seqTy.getEleTy().isa<fir::CharacterType>() &&
+           "entity has no character length");
```

Comment:
The test is for the element type to be `!fir.char` but the message is about length. Maybe "entity is not a character"?